domain/src/main/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonImpl.kt

```diff
@@ -0,0 +1,55 @@
+package org.oppia.android.domain.platformparameter
+
+import org.oppia.android.app.model.PlatformParameter
+import org.oppia.android.util.platformparameter.PlatformParameterSingleton
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Singleton which helps in storing and providing Platform Parameters at runtime. */
+@Singleton
+class PlatformParameterSingletonImpl @Inject constructor() : PlatformParameterSingleton {
+  private var platformParameterMap: Map<String, PlatformParameter> = mapOf()
```

Comment:
This could be a val with a mutable list, instead, where you clear & perform a ``putAll`` to reinitialize it versus changing the parameter itself.

Reply:
Done